### PR TITLE
dnsdist: Gracefully handle a failure to create a TLS server context

### PR DESCRIPTION
**Pull request by rgacogne**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/libssl.cc [hunk 1]
```diff
@@ -806,6 +806,11 @@ std::unique_ptr<SSL_CTX, void(*)(SSL_CTX*)> libssl_init_server_context(const TLS
 {
   auto ctx = std::unique_ptr<SSL_CTX, void(*)(SSL_CTX*)>(SSL_CTX_new(SSLv23_server_method()), SSL_CTX_free);
```
Comment (fredmorcos) [inline pdns/libssl.cc:807]: While we're at it, is it OK to use `decltype(&SSL_CTX_free)` here?

File: pdns/libssl.cc [hunk 2]
```diff
@@ -806,6 +806,11 @@ std::unique_ptr<SSL_CTX, void(*)(SSL_CTX*)> libssl_init_server_context(const TLS
...
 
+  if (!ctx) {
+    ERR_print_errors_fp(stderr);
+    throw std::runtime_error("Error creating an OpenSSL server context");
```
Comment (fredmorcos) [inline pdns/libssl.cc:811]: Is it important to have the errors printed here? If not, then why not use `throw pdns::OpenSSL::error(...)`?

**Review by fredmorcos — APPROVED** [review]

File: pdns/libssl.cc (hunk 2 above)
Comment (rgacogne) [reply pdns/libssl.cc:811]: That would be better indeed, I just have to get used to it :)

File: pdns/libssl.cc (hunk 1 above)
Comment (rgacogne) [reply pdns/libssl.cc]: Done!

**Pull request merged**